src/JustEat.StatsD/IpTransport.cs

```diff
@@ -14,14 +14,26 @@ public IpTransport(IPEndPointSource endPointSource)
             _endpointSource = endPointSource ?? throw new ArgumentNullException(nameof(endPointSource));
         }
 
-        public void Send(string metric)
+        public void Send(ReadOnlySpan<byte> metric)
         {
-            var endpoint = _endpointSource.GetEndpoint();
-            var bytes = Encoding.UTF8.GetBytes(metric);
+            if (metric.Length == 0)
+                return;
 
-            using (var socket = CreateSocket())
+            var endpoint = _endpointSource.GetEndpoint();
+            Socket socket = null;
+            byte[] buffer = null;
+            
+            try
+            {
+                buffer = ArrayPool<byte>.Shared.Rent(metric.Length);
+                metric.CopyTo(buffer);
+                socket = CreateSocket();
+                socket.SendTo(buffer, metric.Length, SocketFlags.None, endpoint);
```

Comment:
There are `Span` overloads of [`Socket.Send`](https://apisof.net/catalog/System.Net.Sockets.Socket.Send(ReadOnlySpan%3CByte%3E,SocketFlags)), but I found they just allocate internally, but I wasn't convinced it wasn't be doing something dumb.

Reply:
Send and SendTo are handled by different winsock methods internally, I did not check any further. There is no overload for SendTo accepting span, but that might get fixed soon.

Reply:
Cool, I wasn't aware of that. I was under the impression the `SendTo` version was a convenience wrapper for the typical use case (see here: https://github.com/justeat/JustEat.StatsD/pull/55/commits/c18e070e5ee44cca6453b01528d04cb7b9babe1a)